tensorflow/core/nccl/nccl_manager.cc

```diff
@@ -703,6 +703,9 @@ void NcclManager::LoopKernelLaunches(NcclStream* nccl_stream) {
         if (p->output) {
           recvbuff = const_cast<char*>(p->output->tensor_data().data());
           num_elements = p->output->NumElements();
+        } else {
+          // Operate in-place if no output (for the src node).
+          recvbuff = const_cast<void*>(sendbuff);
```

Comment:
nit: could you change to `const_cast<char*>` to be consistent with the rest of this file?

Reply:
Both `sendbuff` and `recvbuff` are `[const] void*`, so I don't think this makes sense? (and it doesn't compile). The other cases in the file are casting `tensor_data().data()`, which is a `char*`.

Reply:
Ah yes, I missed that you are casting sendbuff and not tensor_data.data().